### ISSUE TITLE
Advanced Uploads

### ISSUE THREAD

**Issue by thombruce** [thombruce/helvellyn#87]
So we have an "Upload" type added to forms, handled by checking attributes for matching a certain class in a `multipart/form-data` submission... and this is all done on the **EntitiesController**.

But this doesn't help us add our images to content in our Rich Text Editor. That still only supports inclusion via URL... and that will essentially be true forever. What we need there is a mechanism for exchanging an upload for a URL. So... _How's that going to work?_

### Add an UploadsController

This will allow us to create uploads by themselves via the same `multipart/form-data` submissions. We can then modify the _add image_ option in the TipTap editor to perform these uploads and receive a URL.

### Modify TipTap Editor Image Option

This presently just shows a form with three fields for image data, the top of which is an image URL... this should remain a URL by the time submission comes, but we should also allow for image upload.

In other words... click some "attach/upload image" button, select the image from your computer, that will add it to an input. This should then **automatically** upload, and we should receive a URL in response?

_That's fine but... we might spend some time there waiting for a response._

_Maybe we should also base64 encode the image for immediate display. That would be nice, but you then have to replace that with the URL when it's returned, and that means finding the correct item to replace... and handling errors from the server when upload is unsuccessful..._

For the time being, accept the delay and lock the user into upload while they wait - it should be fast in any case. But some solid future considerations there above.
